cbmc/proofs/IotMqtt_IsSubscribed/IotMqtt_IsSubscribed_harness.c

```diff
@@ -0,0 +1,30 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "mqtt_state.h"
+
+void harness()
+{
+  IotMqttConnection_t mqttConnection = allocate_IotMqttConnection(NULL);
+  __CPROVER_assume(mqttConnection);
+  ensure_IotMqttConnection_has_lists(mqttConnection);
+  __CPROVER_assume(valid_IotMqttConnection(mqttConnection));
+
+  allocate_IotMqttSubscriptionList(&(mqttConnection->subscriptionList), 1);
+  __CPROVER_assume(valid_IotMqttSubscriptionList(&(mqttConnection->subscriptionList), 1));
+
+  uint16_t topicFilterLength;
+  __CPROVER_assume(topicFilterLength < TOPIC_LENGTH_MAX);
+
+  char* TopicFilter = malloc_can_fail(topicFilterLength);
```

Comment:
I think there is an invariant assumed in MQTT that (length == 0) iff (string == NULL) for strings and their lengths.  But if this works, that's great.